[telemetrygen] Add batching to metrics and logs and fixes tracing

bogdan-st opened the pull request:
<!--Ex. Fixing a bug - Describe the bug and how this fixes the issue.
Ex. Adding a feature - Explain what this achieves.-->
#### Description
  - Fixed traces batching to have configurable batch size and match batch flag.
  - Added batching to metrics.
  - Added batching to logs. 
Currently only batch-size will trigger sends for metrics and logs. Might add batch timeout in a future PR. The traces one has a 1 second timeout by default that is not configurable, i wanted to expose it when all of them have that capability. 
<!-- Issue number (e.g. #1234) or full URL to issue, if applicable. -->
#### Link to tracking issue
Part of #42322 but does not complete it

<!--Describe what testing was performed and which tests were added.-->
#### Testing
  - Added unit tests and integration tests. 
  Now every test has their own build since sometimes one test would finish before the others and would delete the binary before the others would use it. I could not figure out a way to make them use the same one. 
<!--Describe the documentation added.-->
#### Documentation
  - Changed README.md

bogdan-st committed:
add batch to metrics

bogdan-st committed:
logs batching

bogdan-st committed:
fix traces batch option

bogdan-st committed:
some fixes

atoulme commented:
> Added unit tests and integration tests.
Now every test has their own build since sometimes one test would finish before the others and would delete the binary before the others would use it. I could not figure out a way to make them use the same one.

Please open an issue to follow up.

bogdan-st committed:
merge main

bogdan-st committed:
remove Config prefix

bogdan-st committed:
add more tests

bogdan-st committed:
Merge branch 'main' into batch

bogdan-st committed:
fix readme

bogdan-st committed:
Merge remote-tracking branch 'refs/remotes/origin/batch' into batch

bogdan-st committed:
Merge upstream/main into batch branch

- Resolved conflict in cmd/telemetrygen/pkg/traces/config.go
- Preserved Batch field and all batch-related tests
- Integrated upstream NumSpanLinks feature

bogdan-st committed:
Fix duplicate batch flag definition in traces config

- Remove Batch field from traces config (already in common config)
- Remove duplicate flag registration and default value
- Batch functionality is inherited from common.Config
- All integration tests now pass (22 tests)

bogdan-st committed:
Fix changelog component name

Change 'telemetrygen' to 'cmd/telemetrygen' to match valid component names

mx-psi commented:
I'm supportive of this PR moving forward

bogdan-st committed:
merge

bogdan-st committed:
fix some tests

codeboten committed:
Merge branch 'main' into batch

bogdan-st committed:
Merge branch 'main' into batch

bogdan-st committed:
Merge branch 'main' into batch

codeboten committed:
Merge branch 'main' into batch

mx-psi committed:
Merge branch 'main' into batch
